parser.go

```diff
@@ -158,24 +164,36 @@ func (p *Parser) parseEscapedExpansion() (EscapedExpansion, error) {
 	}
 }
 
-func (p *Parser) parseExpansion() (Expansion, error) {
+func (p *Parser) parseExpansion() (result Expansion, isLiteral bool, err error) {
 	if c := p.nextRune(); c != '$' {
-		return nil, fmt.Errorf("Expected expansion to start with $, got %c", c)
+		err = fmt.Errorf("Expected expansion to start with $, got %c", c)
+		return
 	}
 
+	c := p.peekRune()
+
 	// if we have an open brace, this is a brace expansion
-	if c := p.peekRune(); c == '{' {
-		return p.parseBraceExpansion()
+	if c == '{' {
+		result, err = p.parseBraceExpansion()
+		return
 	}
 
+	// if not a letter, it's a literal dollar sign
+	if !unicode.IsLetter(c) && !unicode.IsNumber(c) {
```

Comment:
I think strings starting with numbers aren't considered to be identifiers (`scanIdentifier` only checks `unicode.IsLetter`)

Reply:
Yeah I only added IsNumber to avoid [this test](https://github.com/buildkite/interpolate/blob/d30645a39e08917571f6cd1e5ab267e88d70c609/interpolate_test.go#L117). Should we raise an error for a number, or just treat as literal? My naive opinion is to treat as literal and not raise.

Reply:
I agree, for our purposes (pipeline interpolation) it's a literal - similar to `$(...)`.